jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/AsyncContentProducer.java

```diff
@@ -395,6 +395,12 @@ void release()
             _condition.signal();
         }
 
+        void fail()
```

Comment:
This whole class lacks javadoc.

Rather than have n >= 3 copies of it, why not move to jetty-util and document properly

Reply:
Added javadoc.
Regarding the 3 copies, EE9 is slightly different but EE10/11 are indeed identical. But this class is very specific to the servlet input FSM so I'd rather move it to a module common between EE* versions rather than moving it to jetty-util as I seriously doubt that it can be reused.